src/pytorch_lightning/loops/epoch/training_epoch_loop.py

```diff
@@ -624,7 +624,7 @@ def _get_max_shape(array: List[Any]) -> List[int]:
         >>> _get_max_shape([[], [[1], [2]], []])
         [3, 2, 1]
     """
-    dimensions = defaultdict(int)
+    dimensions: DefaultDict[Any, int] = defaultdict(int)
```

Comment:
I think they are both ints?

```suggestion
    dimensions: DefaultDict[int, int] = defaultdict(int)
```

Reply:
Ah, you're right. `level` returned by `_get_dimensions` is `int`. Thank you!